### PR TITLE
fix(mayastor): add support of uuid and name as separate replica proprties

### PR DESCRIPTION
**Pull request by mtzaurus**
fix(mayastor): add support of uuid and name as separate replica proprties

Added new gRPC methods to support separately UUID and name for replica
creation and enumeration.

Fixes CAS-1036

### CODE REVIEW

File: rpc/proto/mayastor.proto [hunk 1]
```diff
@@ -163,11 +174,27 @@ message Replica {
   string uri = 6;   // uri usable by nexus to access it
 }
 
+// Replica2 properties
```
Comment (jonathan-teh) [inline rpc/proto/mayastor.proto:177]: ReplicaV2

File: test/python/test_replica_uuid.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
+import pytest
```
Comment (jonathan-teh) [inline test/python/test_replica_uuid.py:1]: Please add this file to `scripts/pytest-tests.sh`.

**Review by jonathan-teh — APPROVED** [review]
LGTM, "properties" is misspelled in the first comment.

**Review by tiagolobocastro — APPROVED** [review]

File: mayastor/src/bin/mayastor-client/replica_cli.rs [hunk 3]
```diff
@@ -47,6 +46,44 @@ pub fn subcommands<'a, 'b>() -> App<'a, 'b> {
...
+                    Arg::with_name("uuid")
+                        .required(true).index(3)
+                        .help("Unique replica uuid"))
+                .arg(
+                    Arg::with_name("protocol")
+                        .short("p")
+                        .long("protocol")
+                        .takes_value(true)
+                        .value_name("PROTOCOL")
+                        .help("Name of a protocol (nvmf, iSCSI) used for sharing the replica (default none)"))
```
[the remark this answers is not in the document]
Comment (jonathan-teh) [reply mayastor/src/bin/mayastor-client/replica_cli.rs:71]: Hmm... the iSCSI initiator has been removed so the nexus can't connect to an iSCSI replica but you can still share a replica over iSCSI. Perhaps we need another ticket to remove that.

File: mayastor/src/bin/mayastor-client/replica_cli.rs [hunk 4]
```diff
@@ -47,6 +46,44 @@ pub fn subcommands<'a, 'b>() -> App<'a, 'b> {
...
+                .arg(
+                    Arg::with_name("size")
+                        .short("s")
+                        .long("size")
+                        .takes_value(true)
+                        .required(true)
+                        .value_name("NUMBER")
+                        .help("Size of the replica"))
+                .arg(
```
[the remark this answers is not in the document]
Comment (mtzaurus) [reply mayastor/src/bin/mayastor-client/replica_cli.rs:80]: Thin provisioning is gonna be fixed, yes. CAS-721 has been reopened, so we might have to change CLI arguments too.

**Pull request merged**